packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -619,7 +619,12 @@ export class PeerManager {
         }
       },
       {retries: 3, retryDelay: 1000}
-    ).catch((err) => {
+    ).catch((err: Error) => {
+      if (err.message.includes("NotFound")) {
+        return this.logger.debug("Peer not found when attempting to set agentVersion", {
+          peerId: peerData.peerId.toString(),
+        });
+      }
       this.logger.error("Error setting agentVersion for the peer", {peerId: peerData.peerId.toString()}, err);
```

Comment:
As I understood Tuyen and based on my own understanding of this error, we should move this log to debug. I honestly do not think we need to handle the NotFound error separately in this case, especially as it seems to be a more recent bug and might be resolved soon

```suggestion
    ).catch((err) => {
      this.logger.debug("Error setting agentVersion for the peer", {peerId: peerData.peerId.toString()}, err);
```